### PR TITLE
Decorator

### PR DESCRIPTION
**Pull request by joskalenda**
## OOP school library: decorate a class

- [ ] Learning objectives

- Implement composition.
- Run a program using the command line.

### Features

- [ ] Create a class Nameable.

- Implement a method called correct_name that will raise a NotImplementedError.
- Turn your Person class to Nameable
- Make sure that your Person class inherits from Nameable
- Make sure that this class has a method correct_name implemented. It should simply return the name attribute.

- [ ] Create a class Decorator

- Make sure that it inherits from Namable.
- In the constructor assign a namable object from params to an instance variable.
- Implement the correct_name method that returns the result of the correct_name method of the @namable.

- [ ] Create class CapitalizeDecorator and TrimmerDecorator

1. For the CapitalizeDecorator:

- Create a class that inherits from the base Decorator class.
- Implement a method correct_name that capitalizes the output of @nameable.correct_name.

2. For the TrimmerDecorator:

- Create a class that inherits from the base Decorator class.
- Implement a method correct_name that makes sure that the output of @nameable.correct_name has a maximum of 10 characters. If it's longer it should trim the word.

#### Result

> || Test case ||
> |-|-----|-|
> || rub ||

### CODE REVIEW

File: decore.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
+class Nameable
+  def correct_name
+    raise 'NotImplementedError'
+  end
+end
```
Comment (mateo951) [inline decore.rb:5]: Well done! You create Namable class with a method 'correct_name' that raises an error 🔥

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
+require './decore'
+
+class Person < Nameable
```
Comment (mateo951) [inline person.rb:3]: Nice 🔥 Your Person class inherits from Nameable!

File: decore.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+class Decore < Nameable
+  attr_accessor :nameable
+
+  def initialize(nameable)
```
Comment (mateo951) [inline decore.rb:10]: Namable object from params is assigned to an instance variable 😎

**Review by mateo951 — APPROVED** [review]
### Hi @joskalenda, 

📝👌 **Your project highlights** 🔥😎 
- General Requirements ✔
- Fantastic PR 🤓
- Decorator implementation 😉

Your project is complete!  There is nothing else to say other than... it's time to merge it ✔ 
### Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by joskalenda** [comment]
Thanks so much, @mateo951, I appreciate your comments 😄  .

**Pull request merged**